corehq/apps/accounting/forms.py

```diff
@@ -566,7 +566,7 @@ def clean(self):
             and start_date > self.cleaned_data['end_date']):
             raise ValidationError("End date must be after start date.")
 
-        if self.cleaned_data['end_date'] <= datetime.date.today():
+        if self.cleaned_data['end_date'] and self.cleaned_data['end_date'] <= datetime.date.today():
```

Comment:
Also your "local timezone" phrase is ambiguous. There are probably at least two local timezones that come into play here:

- the server's local timezone (`datetime.date.today()`)
- the user/browser local timezone (presumably `end_date`, although I didn't track it down to verify where that date originated)